src/plugins/enum/testmod_enum.c

```diff
@@ -0,0 +1,76 @@
+/**
+* \file
+*
+* \brief Tests for enum plugin
+*
+* \copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+*
+*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <kdbconfig.h>
+#include <stdio.h>
+
+#include <tests_plugin.h>
+#include <tests_internal.h>
+
+
+static void test()
+{
+	Key *parentKey = keyNew("user/tests/enum", KEY_VALUE, "", KEY_END);
+	Key *k1 = keyNew("user/tests/enum/valid1", KEY_VALUE, "TRUE", KEY_META, "check/enum", "[\'TRUE\',\'FALSE\']", KEY_END);
+	Key *k2 = keyNew("user/tests/enum/valid2", KEY_VALUE, "FALSE", KEY_META, "check/enum", "[\'TRUE\',\'FALSE\']", KEY_END); 
+	Key *k3 = keyNew("user/tests/enum/invalid1", KEY_VALUE, "BLA", KEY_META, "check/enum", "[\'TRUE\',\'FALSE\']", KEY_END);
+	Key *k4 = keyNew("user/tests/enum/invalid2", KEY_VALUE, "", KEY_META, "check/enum", "[\'TRUE\',\'FALSE\']", KEY_END);
+	KeySet *conf = ksNew(0, KS_END);
+	KeySet *ks;
+	PLUGIN_OPEN("enum");
+	
+	ks = ksNew(20, KS_END); 
+	ksAppendKey(ks, k1);
+	ksRewind(ks);
+	succeed_if(plugin->kdbSet(plugin, ks, parentKey) == (1), "kdbSet failed");
```

Comment:
can you write a function/macro for these 5 lines. Would be better if every test is a one-liner.